olive/systems/azureml/aml_system.py

```diff
@@ -76,6 +76,7 @@ def get_asset_type_from_resource_path(resource_path: ResourcePath):
 
 class AzureMLSystem(OliveSystem):
     system_type = SystemType.AzureML
+    olive_config = None
```

Comment:
Not sure if creating this as a class attribute is a good idea since changing the value for one instance would change for other instances too. I think class attributes are generally immutable? Probably not an issue in Olive since we run workflows from the command line but still feels weird to use a class attribute here.

What about the aml system initializes with an `None` instance attribute `olive_config`. And the engine uses a method like `set_olive_config` to update the instance's value?

Reply:
There could in theory be two workflows running in parallel with aml systems. The second one would update the attribute for the first one too.

Reply:
I saw that @guotuofeng suggested this change. What is you thought on this?

Reply:
I'd like not change the OliveSystem interface to only support the scenario for AML. 

If you have concerns on class level, then we can move this kind of logic like what Xiaoyu previous changes. 
After creating the host/target, if it is aml then we can update there. The only issue with that approach is that the engine need hold the entire config even if it doesn't use it. 

Since we will have only one engine run at the same time, I give the suggest. 

As for two workflows, do you mean in one process there will be two different engine configs?

Reply:
I think it is okay for now since the hypothetical use case I described doesn't happen. If a process is running two workflows sequentially, then it's not an issue. Only becomes an issue if they run in parallel or the second one starts before the second one. 

For the sake of simplicity, I think the current idea is fine. We can revisit in the future if any issue arises.